akka-http-core/src/main/scala/akka/http/scaladsl/model/headers/headers.scala

```diff
@@ -36,6 +36,14 @@ sealed abstract class ModeledCompanion[T: ClassTag] extends Renderable {
       case res                                         ⇒ Left(res.errors)
     }
 }
+/** INTERNAL API */
+private[akka] object ModeledCompanion {
+  def nameFromClass[T](clazz: Class[T]): String = {
+    val name = clazz.getSimpleName.replace("$minus", "-")
```

Comment:
getSimpleName has problems in Scala.

Use: https://github.com/akka/akka/blob/master/akka-actor/src/main/scala/akka/event/Logging.scala#L386